modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -996,6 +996,171 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
+                               float scale = 1.0, bool swapRB = false, bool crop = false);
+
+         /** @brief Given the @p input frame, create input blob, run net and return the output @p blobs.
+          *  @param[in]  frame  The input image.
+          *  @param[out] outs Allocated output blobs, which will store results of the computation.
+          */
+         CV_WRAP void predict(InputArray frame, CV_OUT OutputArrayOfArrays outs);
+
+         /** @brief The static method creates model from config and weights.
+         * @param[in] model Binary file contains trained weights.
+         * @param[in] config Text file contains network configuration.
+        */
+         CV_WRAP static Ptr<Model> create(const String& model, const String& config = "");
+
+     protected:
+         struct Impl;
+         Ptr<Impl> impl;
+     };
+
+     /** @brief This class is presented high-level API for classification models.
+      *
+      * ClassificationModel allows to set params for preprocessing input image.
+      * ClassificationModel creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return top-1 prediction.
+      */
+     class CV_EXPORTS_W ClassificationModel : public Model
+     {
+     public:
+         /**
+          * @brief Create classification model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+         ClassificationModel(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from deep learning network.
+          * @param[in] network Net object.
+          */
+         ClassificationModel(const Net& network);
+
+         /** @brief Given the @p input frame, create input blob, run net and return top-1 prediction.
+          *  @param[in]  frame  The input image.
+          */
+         std::pair<int, float> classify(InputArray frame);
+
+         /** @overload */
+         CV_WRAP void classify(InputArray frame, CV_OUT int& classId, CV_OUT float& conf);
+
+         /** @brief The static method creates classification model from config and weights.
+         * @param[in] model Binary file contains trained weights.
+         * @param[in] config Text file contains network configuration.
+        */
+         CV_WRAP static Ptr<ClassificationModel> create(const String& model, const String& config = "");
+     };
+
+     /** @brief This class is presented high-level API for detection models.
+      *
+      * DetectionModel allows to set params for preprocessing input image.
+      * DetectionModel creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return result detections.
+      */
+     class CV_EXPORTS_W DetectionModel : public Model
```

Comment:
I think for certain models such DetectionModel or ClassificationModel we need just brief summary.